src/conversion/modelconversiondispatcher.js

```diff
@@ -250,7 +252,9 @@ export default class ModelConversionDispatcher {
 				item: item
 			};
 
-			this._testAndFire( 'remove', data, consumable );
+			const name = removeAs || item.name || '$text';
+
+			this._testAndFire( `remove`, consumable, data, name );
```

Comment:
If we go with your solution, the "fake" old item has to be bind with original element in view (so mapper correctly maps ranges in remove converter). It was like this in old implementation, when "rename converter" was in `model-to-view-converters.js`. I will check how complicated the code will end up with this solution.